### ISSUE TITLE
Error: self signed certificate in certificate chain

### ISSUE THREAD

**Issue by dageniusal** [OmniSharp/omnisharp-vscode#930]
## Environment data
.NET Command Line Tools (1.0.0-preview2-003133)

Product Information:
 Version:            1.0.0-preview2-003133
 Commit SHA-1 hash:  74df06500c

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

C# Extension version:
1.5.0

## Steps to reproduce
Reload VS Code after upgrading extension or upon startup

## Expected  behavior
Ignore self-signed certificates according to the "http.proxyStrictSSL" setting.

## Actual behavior
Download does not follow the environment settings and checks for Self-Signed Certificates, which causes installation to fail.

**Comment by DustinCampbell** [comment]
Looks like we ignore when this is set to false here: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/proxy.ts#L35

**Comment by DustinCampbell** [comment]
I just shipped a new 1.5.1 release on the marketplace with a fix for this issue. Could you let me know if it works better for you?

**Comment by dageniusal** [comment]
Unfortunately, it doesn't look like it resolved the issue.

    Updating C# dependencies...
    Downloading package 'OmniSharp (.NET 4.6 / x64)' (12309 KB) .................... Done!
    Downloading package '.NET Core Debugger (Windows / x64)' Failed at stage: downloadPackages
    Error: self signed certificate in certificate chain

    Finished
    [ERROR]: C# Extension failed to install the debugger package

settings.json is as follows:

    // Place your settings in this file to overwrite the default settings
    {
        "http.proxyStrictSSL": false
    }

**Comment by DustinCampbell** [comment]
+ @gregg-miskelly and @chuckries: It looksl ike this is failing specifically in the debugger stage. Signing issue?

**Comment by gregg-miskelly** [comment]
Interesting. No idea what is going on, but we will have a look and see what we can find.

**Comment by gregg-miskelly** [comment]
@dageniusal Can you try a quick experiment?

1. Open up %USERPROFILE%\.vscode\extensions\ms-vscode.csharp-1.5.0\package.json in VS Code (or some other text editor)
2. Go to ~line 141 -- should be: "url": "https://vsdebugger.azureedge.net/coreclr-debug-1-5-0/coreclr-debug-win7-x64.zip",
3. Change this URL to https://vsdebugger.blob.core.windows.net/coreclr-debug-1-5-0/coreclr-debug-win7-x64.zip
4. Restart VS Code and see if this URL succeeds

**Comment by dageniusal** [comment]
@gregg-miskelly I manually updated the package.json file as requested and the update was successful. Thanks a million!

**Comment by federicobarera** [comment]
ah, same issue, same time.
Tried the alternative url and worked fine

**Comment by gregg-miskelly** [comment]
Thanks @federicobarera and @dageniusal! We will try to push out a new release tomorrow with this fix.

**Comment by DustinCampbell** [comment]
@gregg-miskelly: Be sure to prepare the fix for this against the release branch.

**Comment by DustinCampbell** [comment]
We've prepared a new release that we hope will fix this issue. Could you try it out and ensure that it works for you? Use the following steps to download and install the new version of the extension:

1. Download the VSIX from [here](https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.5.2)
2. Open Visual Studio Code
3. Select View->Extensions to display the Extensions pane.
4. In the Extensions pane, click the little gear icon next to the C# extension and select `Uninstall`.
5. Confirm any prompts and allow Visual Studio Code to reload itself.
6. In the Extensions pane, click the [...] that appears in the top-right corner and select "Install from VSIX..."
7. Select the VSIX you downloaded.
8. Let Visual Studio Code reload itself.

**Comment by federicobarera** [comment]
It seems fine. 
After the installation it shown an alert that the program it's trying to use .net framework 4.6 which I don't have installed. I am installing it now to see if it runs fine.

Updating C# dependencies...
Platform: win32, x86_64 (win7-x64)

Downloading package 'OmniSharp (.NET 4.6 / x64)' (12310 KB) .................... Done!
Downloading package '.NET Core Debugger (Windows / x64)' 	Retrying from 'https://vsdebugger.blob.core.windows.net/coreclr-debug-1-5-0/coreclr-debug-win7-x64.zip' (21295 KB) .................... Done!

Installing package 'OmniSharp (.NET 4.6 / x64)'
Installing package '.NET Core Debugger (Windows / x64)'

Finished

**Comment by DustinCampbell** [comment]
Excellent. That last bit to install .NET Framework 4.6 was just added.

**Comment by federicobarera** [comment]
Yeah, after installing .net 4.6 and re-installing the extensions all goes smoothly.
Great job, thanks.

**Comment by DustinCampbell** [comment]
w00t!

**Comment by dageniusal** [comment]
@DustinCampbell, I followed the instructions as directed and it installed correctly. I already had .NET 4.6.* installed on my machine, so I didn't need to take the same action as @federicobarera.

Thanks for the great work!

```
Updating C# dependencies...
Platform: win32, x86_64 (win7-x64)

Downloading package 'OmniSharp (.NET 4.6 / x64)' (12310 KB) .................... Done!
Downloading package '.NET Core Debugger (Windows / x64)' 	Retrying from 'https://vsdebugger.blob.core.windows.net/coreclr-debug-1-5-0/coreclr-debug-win7-x64.zip' (21295 KB) .................... Done!

Installing package 'OmniSharp (.NET 4.6 / x64)'
Installing package '.NET Core Debugger (Windows / x64)'

Finished
```

**Comment by DustinCampbell** [comment]
Awesome!

**Comment by netcore-jroger** [comment]
why ?
-------------------------------------------------------------------
Updating C# dependencies...
Platform: win32, x86_64 (win7-x64)

Downloading package 'OmniSharp (.NET 4.6 / x64)' (12310 KB) .................... Done!
Downloading package '.NET Core Debugger (Windows / x64)' (21295 KB) . Done!

Installing package 'OmniSharp (.NET 4.6 / x64)'
Installing package '.NET Core Debugger (Windows / x64)'
Failed at stage: installPackages
**Error: end of central directory record signature not found**

Finished
[ERROR]: C# Extension failed to install the debugger package

**Comment by gregg-miskelly** [comment]
@netcore-jroger what version of the C# extension do you have?

**Comment by DustinCampbell** [comment]
It looks to me like the debugger bits didn't completely download so the ZIP file was likely corrupt.

**Comment by DustinCampbell** [comment]
FWIW, this is a very different issue than the one reported in this bug. It might be worth submitting a new issue rather than commenting on a closed one.

**Comment by netcore-jroger** [comment]
@gregg-miskelly  C# version :  1.5.2

**Comment by netcore-jroger** [comment]
moved to [https://github.com/OmniSharp/omnisharp-vscode/issues/977](https://github.com/OmniSharp/omnisharp-vscode/issues/977)

**Comment by psoszynski** [comment]
Today, I got the same thing:

Updating C# dependencies...
Platform: win32, x86 (win7-x86)

Downloading package 'OmniSharp (.NET 4.6 / x86)' (12181 KB) .................... Done!

Installing package 'OmniSharp (.NET 4.6 / x86)'

Finished
[ERROR]: C# Extension failed to install the debugger package

Then I followed the instructions from this thread, installed C# extension (1.5.2) from "VSIX", restarted VSCode.

But the error is still there.
